src/index.test.mjs

```diff
@@ -65,62 +65,171 @@ test("astroRehypeRelativeMarkdownLinks", async (t) => {
+        .process(input);
+
+      const expected =
+        '<html><head></head><body><a href="./fixtures/does-not-exist.md">foo</a></body></html>';
+
+      assert.equal(actual, expected);
+    },
+  );
+
+  await t.test(
+    "should transform file paths that exist with non alphanumeric characters",
+    async () => {
+      const input = '<a href="./fixtures/test%20(non-alpha).md">foo</a>';
+      const { value: actual } = await rehype()
+        .use(testSetupRehype)
+        .use(astroRehypeRelativeMarkdownLinks, { contentPath: "src" })
+        .process(input);
+
+      const expected =
+        '<html><head></head><body><a href="/fixtures/test-non-alpha">foo</a></body></html>';
+
+      assert.equal(actual, expected);
+    },
+  );
+
+  await t.test("should not replace absolute path if file exists", async () => {
+    const absolutePath = path.resolve("./fixtures/test.md");
+    const input = `<a href="${absolutePath}">foo</a>`;
+    const { value: actual } = await rehype()
+      .use(testSetupRehype)
+      .use(astroRehypeRelativeMarkdownLinks, { contentPath: "src" })
+      .process(input);
+
+    const expected = `<html><head></head><body><a href="${absolutePath}">foo</a></body></html>`;
+
+    assert.equal(actual, expected);
+  });
+
+  await t.test(
+    "should not replace absolute path if file does not exist",
+    async () => {
+      const absolutePath = `${path.dirname("./fixtures/test.md")}/does-not-exist.md`;
+      const input = `<a href="${absolutePath}">foo</a>`;
+      const { value: actual } = await rehype()
+        .use(testSetupRehype)
+        .use(astroRehypeRelativeMarkdownLinks, { contentPath: "src" })
+        .process(input);
+
+      const expected = `<html><head></head><body><a href="${absolutePath}">foo</a></body></html>`;
+
+      assert.equal(actual, expected);
+    },
+  );
+
+  await t.test(
+    "should not transform index.md file paths if file does not exist",
+    async () => {
+      const input = '<a href="./fixtures/does-not-exist/index.md">foo</a>';
+      const { value: actual } = await rehype()
+        .use(testSetupRehype)
+        .use(astroRehypeRelativeMarkdownLinks, { contentPath: "src" })
+        .process(input);
+
+      const expected =
+        '<html><head></head><body><a href="./fixtures/does-not-exist/index.md">foo</a></body></html>';
+
+      assert.equal(actual, expected);
+    },
+  );
+
+  await t.test("should not replace path if .md directory exists", async () => {
+    const input = '<a href="./fixtures/dir-exists.md">foo</a>';
     const { value: actual } = await rehype()
       .use(testSetupRehype)
       .use(astroRehypeRelativeMarkdownLinks, { contentPath: "src" })
       .process(input);
 
     const expected =
-      '<html><head></head><body><a href="/fixtures/test?q=q#hash">foo</a></body></html>';
+      '<html><head></head><body><a href="./fixtures/dir-exists.md">foo</a></body></html>';
 
     assert.equal(actual, expected);
   });
```

Comment:
For tests like these, I'm wondering if we should transform them as the directory exists, even though an index-esque file doesn't exist. But, I thinks it's likely fine as it would be new functionality. Not sure if it would match what Astro is doing

Reply:
Interesting though, I hadn't considered that.  My thought is that the library requirement should be that the file path always points to a file and we don't make any assumptions/checks for whether or not an index file exists - it could be `index.md`, `index.mdx`, they could have a custom slug for default file in a directory (via a custom `getStaticPaths` method) - I think too many possible scenarios to cover.  If we require the link always points to a physical file that exists or we bail, I think we're safe that way.  Thoughts?

Reply:
> we require the link always points to a physical file that exists or we bail, I think we're safe that way.

Yeah, that sounds like the right way to deal with this. Thanks for keeping me honest